extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -827,6 +831,166 @@ export class ProjectsController {
 		return result;
 	}
 
+	public async selectAutorestSpecFile(): Promise<string | undefined> {
+		let quickpickSelection = await vscode.window.showQuickPick(
+			[constants.browseEllipsis],
+			{ title: constants.selectSpecFile, ignoreFocusOut: true });
+		if (!quickpickSelection) {
+			return;
+		}
+
+		const filters: { [name: string]: string[] } = {};
+		filters['OpenAPI/Swagger spec'] = ['yaml'];
+
+		let uris = await vscode.window.showOpenDialog({
+			canSelectFiles: true,
+			canSelectFolders: false,
+			canSelectMany: false,
+			openLabel: constants.selectString,
+			filters: filters,
+			title: constants.selectSpecFile
+		});
+
+		if (!uris) {
+			return;
+		}
+
+		return uris[0].fsPath;
+	}
+
+	/**
+	 * @returns \{ newProjectFolder: 'C:\Source\MyProject',
+	 * 			outputFolder: 'C:\Source',
+	 * 			projectName: 'MyProject'}
+	 */
+	public async selectAutorestProjectLocation(specPath: string): Promise<{ newProjectFolder: string, outputFolder: string, projectName: string } | undefined> {
+		let valid = false;
+		let newProjectFolder: string = '';
+		let outputFolder: string = '';
+		let projectName: string = '';
+
+		let quickpickSelection = await vscode.window.showQuickPick(
+			[constants.browseEllipsis],
+			{ title: constants.selectProjectLocation, ignoreFocusOut: true });
+		if (!quickpickSelection) {
+			return;
```

Comment:
for other operations (publish and script) the UI part of the code is in a different file unde UI folder. should we keep the pattern?

Reply:
Oh!  I must've missed this change.  I'm on the fence with that.

Originally, we had a controller that did the logic, and the separate file just for the GUI that then triggered the callback (in the controller) to effect the changes once it had collected all of the info.  With @Charles-Gagnon's changes to also have quickpick UIs for VS Code, the line between logic and UI got blurred, resulting in a bunch of logic getting shifted over to the UI folder, which I'm kinda iffy on.

I'm going to leave it as-is for now, but spin up a thread with the team on this topic.